### PR TITLE
Adding extra data-layer params in payload and params dict

### PR DESCRIPTION
**Pull request by vorjdux**

### CODE REVIEW

**Comment by naazgull** [comment]
Thank you for your contribution. 
You shouldn't test if "params" or "payload" are instantiated. You want to pass on the extra parameters even they are not instantiated.

File: python/src/zapata/rest/__init__.py [hunk 1]
```diff
@@ -91,14 +91,21 @@ def dispatch(self, method, topic, envelope, context):
 
         endpoint, variables = self.path_discover(self.web_endpoint, topic, self.data_layer_endpoint)
 
+        params = envelope.get('params')
```
Comment (naazgull) [inline python/src/zapata/rest/__init__.py]: params = zpt.merge(envelope.get('params'), self.variable_resolver(self.data_layer_params, variables)

File: python/src/zapata/rest/__init__.py [hunk 2]
```diff
@@ -91,14 +91,21 @@ def dispatch(self, method, topic, envelope, context):
...
+        payload = envelope.get('payload')
```
Comment (naazgull) [inline python/src/zapata/rest/__init__.py]: payload = zpt.merge(envelope.get('payload'), self.variable_resolver(self.data_layer_params, variables)

File: python/src/zapata/rest/__init__.py [hunk 3]
```diff
@@ -91,14 +91,21 @@ def dispatch(self, method, topic, envelope, context):
...
+
+        if params: params = zpt.merge(params, self.variable_resolver(self.data_layer_params, variables))
```
Comment (naazgull) [inline python/src/zapata/rest/__init__.py]: No need for this

File: python/src/zapata/rest/__init__.py [hunk 4]
```diff
@@ -91,14 +91,21 @@ def dispatch(self, method, topic, envelope, context):
...
+        if payload: payload = zpt.merge(payload, self.variable_resolver(self.data_layer_params, variables))
```
Comment (naazgull) [inline python/src/zapata/rest/__init__.py]: No need for this

File: python/src/zapata/rest/__init__.py (hunk 1 above)
Comment (vorjdux) [reply python/src/zapata/rest/__init__.py]: The bridge doesn't accept merge an empty object with a dictionary, so I needed to test before to have sure that there is an object before to merge.

Comment (naazgull) [reply python/src/zapata/rest/__init__.py]: Bad design, sorry. The merge function MUST accept nil values on either one of the parameters.
Already corrected, will be include in the next release

**Comment by vorjdux** [comment]
Done. Code corrected while we wait the next |apata release with TODOs to remove the if conditions

**Pull request merged**